lib/AccessStaticPropertyWithinModelContextRule.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SlamPhpStan;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\StaticPropertyFetch;
+use PHPStan\Analyser\Scope;
+use PHPStan\Broker\Broker;
+use PHPStan\Rules\Rule;
+
+final class AccessStaticPropertyWithinModelContextRule implements Rule
+{
+    /**
+     * @var Broker
+     */
+    private $broker;
+
+    /**
+     * @var string
+     */
+    private $modelBaseClassOrInterface;
+
+    /**
+     * @var string
+     */
+    private $singletonAccessor;
+
+    public function __construct(Broker $broker, string $modelBaseClassOrInterface, string $singletonAccessor)
+    {
+        $this->broker                    = $broker;
+        $this->modelBaseClassOrInterface = $modelBaseClassOrInterface;
+        $this->singletonAccessor         = $singletonAccessor;
+    }
+
+    public function getNodeType(): string
+    {
+        return StaticPropertyFetch::class;
+    }
+
+    /**
+     * @param \PhpParser\Node\Expr\StaticPropertyFetch $node
+     * @param \PHPStan\Analyser\Scope                  $scope
+     *
+     * @return string[] errors
+     */
+    public function processNode(Node $node, Scope $scope): array
+    {
+        if (! $node->class instanceof Node\Name || ! $node->name instanceof Node\VarLikeIdentifier) {
+            return [];
+        }
+
+        if (! $scope->isInClass()) {
+            return [];
+        }
+
+        $classReflection = $scope->getClassReflection();
+        if (null === $classReflection || ! $classReflection->isSubclassOf($this->modelBaseClassOrInterface)) {
+            return [];
+        }
+
+        $baseYiiClassName = (string) $node->class;
```

Comment:
You should call `$scope->resolveName()` on `$baseYiiClassName` so that it solves stuff like `self` or `parent`.

Reply:
Thank you :+1: